advanced/Scripts/piholeLogFlush.sh

```diff
@@ -8,17 +8,38 @@
 # This file is copyright under the latest version of the EUPL.
 # Please see LICENSE file for your rights under this license.
 
-echo -n "::: Flushing /var/log/pihole.log ..."
-# Test if logrotate is available on this system
-if command -v /usr/sbin/logrotate >/dev/null; then
-  # Flush twice to move all data out of sight of FTL
-  /usr/sbin/logrotate --force /etc/pihole/logrotate; sleep 3
-  /usr/sbin/logrotate --force /etc/pihole/logrotate
+if [[ "$@" != *"quiet"* ]]; then
+  echo -n "::: Flushing /var/log/pihole.log ..."
+fi
+if [[ "$@" == *"once"* ]]; then
+  # Nightly logrotation
+  if command -v /usr/sbin/logrotate >/dev/null; then
+    # Logrotate once
+    /usr/sbin/logrotate --force /etc/pihole/logrotate
+  else
+    # Copy pihole.log over to pihole.log.1
+    # and empty out pihole.log
+    # Note that moving the file is not an option, as
+    # dnsmasq would happily continue writing into the
+    # moved file (it will have the same file handler)
+    cp /var/log/pihole.log /var/log/pihole.log.1
+    echo " " > /var/log/pihole.log
+  fi
 else
-  # Flush both pihole.log and pihole.log.1 (if existing)
-  echo " " > /var/log/pihole.log
-  if [ -f /var/log/pihole.log.1 ]; then
-    echo " " > /var/log/pihole.log.1
+  # Manual flushing
+  if command -v /usr/sbin/logrotate >/dev/null; then
+    # Logrotate twice to move all data out of sight of FTL
+    /usr/sbin/logrotate --force /etc/pihole/logrotate; sleep 3
+    /usr/sbin/logrotate --force /etc/pihole/logrotate
+  else
+    # Flush both pihole.log and pihole.log.1 (if existing)
+    echo " " > /var/log/pihole.log
+    if [ -f /var/log/pihole.log.1 ]; then
+      echo " " > /var/log/pihole.log.1
+    fi
```

Comment:
I guess really it doesn't matter as all it's doing is creating an initial file. So I assume any subsequent entry would be >> and thus show up after the hidden character.

Maybe it's better to do a time stamp or something to  reflect when the log was rotated.

Something like:

echo "Log rotated $(date)" > /var/log/pihole.log